units/provisioning_setup_host.go

```diff
@@ -772,7 +770,10 @@ func attachVolume(ctx context.Context, env evergreen.Environment, h *host.Host)
 		// attach to the host
 		attachment := host.VolumeAttachment{VolumeID: volume.ID, IsHome: true}
 		if err = cloudMgr.AttachVolume(ctx, h, &attachment); err != nil {
-			return errors.Wrapf(err, "can't attach volume '%s' to host '%s'", volume.ID, h.Id)
+			// another job has already attached the volume
+			if !strings.Contains(err.Error(), "IncorrectState") {
+				return errors.Wrapf(err, "can't attach volume '%s' to host '%s'", volume.ID, h.Id)
```

Comment:
arguably we should debug log and then return nil? we wouldn't want provisioning to fail because of this, and we expect that if the device we're trying to attach is already attached then things are fine. 

I'm also a little worried about string parsing error messages, though I think the alternatives are too heavyweight.

Reply:
What are your thoughts about the host being marked provisioned before the first job completes? (https://github.com/evergreen-ci/evergreen/pull/3560#issuecomment-624840371)